CONTRIBUTING.md

```diff
@@ -381,7 +381,11 @@ This project follows the [Kubernetes API conventions](https://github.com/kuberne
 * If a field is required (for our controllers to work) and has a default value specified via OpenAPI schema, but we don't want to force users to set the field, we have to mark the field as optional. Otherwise, the client-side kubectl OpenAPI schema validation will force the user to set it even though it would be defaulted on the server-side.
 
 Optional fields have the following properties:
-* An optional field MUST be marked with `+optional` and include an `omitempty` JSON tag.
+* An optional field MUST be marked with `+optional`.
+* In most cases, optional fields should also have the `omitempty` struct tag (the omitempty option 
+  specifies that the field should be omitted from the JSON encoding if the field has an empty value).
+  However, if you want to have different logic for an optional field which is not provided vs. provided
+  with empty values, do not use omitempty.
```

Comment:
It will be great if we can provide an example here...

Reply:
I dropped this part. I was just about to explain why this makes sense with MachineDeployment replicas. But apparently the combination +optiona, omitempty + pointer works.